40.md

```diff
@@ -0,0 +1,53 @@
+NIP-40
+======
+
+Expiration Timestamp
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+The `expiration` tag enables users to specify a timestamp at UTC-0 at which the message should be considered expired and deleted by relays (or just ignored by clients). This can be useful for making temporary announcements, making limited-time offers, or sending ephemeral messages but should not be seen as a security solution for the content of your messages.
+So be careful when sharing confidential information.
+
+#### Spec
+
+```
+tag: expiration
+options:
+ - [UNIX timestamp in seconds]: required
+```
+
+#### Example
+
+```json
+{
+    "pubkey": "<pub-key>",
+    "created_at": 1000000000,
+    "kind": 1,
+    "tags": [
+      ["expiration", 1600000000]
+    ],
```

Comment:
Good idea the ttl, however, the advantage of the timestamp here is that the relay can easily locate and delete / not send or treat differently a message with an outdated timestamp.
It is also possible to do this by adding the timestamp of the creation date + the ttl but I am for a Nostr philosophy where relays are very simple programs and clients are explicit in their requests.

It would avoid a relay to make a big loop + calculation in the messages to delete them for example.